### PR TITLE
import artist photos from spotify

### PR DESCRIPTION
**Pull request by aaronkim218**
localhost:8080/spotify/clientCreds/import/artist-details

endpoint is working. call the above route

## Description

[Link to Ticket](insert the link to your ticket inside the parenthesis here)

## Checklist:
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] Some tests are included that test my code

### CODE REVIEW

File: backend/internal/service/handler/spotify/import_artist_details.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		return err
+	}
+
+	var updatedIds []int
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+
+	for _, artist := range artists {
+		if len(artist.Images) == 0 {
```
Comment (ddusichka) [inline backend/internal/service/handler/spotify/import_artist_details.go:32]: okay so this PR looks good and i was trying to populate the whole DB by calling the endpoint over and over. it worked great for a while but towards the end, the number of IDs being returned kept shrinking despite there still being hundreds of artists with photos = `''`. eventually, it started just returning null even though some of the remaining artists definitely have spotify images.

I'm pretty sure this comes from this check. ik that if artists don't have spotify photos there's not much we can do. however, i feel like we should stop checking them in that case. two potential solutions i thought of unless you have something better:

1. i'm not sure why we're currently setting photo to `''` on artist creation but maybe we stop doing that and use null properly. and then if we try this check and don't find anything, THEN we set it to `''` and stop checking next time?

OR

2.  (probably better design and would handle the case where an artist magically uploads spotify images later on). we have the SQL not limit at 50 and instead have this method make multiple calls to spotify in batches of 50. that way we're always looking at all of them and even if there are still no-spotify-image artists, they won't prevent valid artist images from being found

hopefully that makes sense but lmk if anything's confusing

**Review by ddusichka — COMMENTED** [review]
looks good! just one comment + lint pls

File: backend/internal/service/handler/spotify/import_artist_details.go (hunk 1 above)
Comment (aaronkim218) [reply backend/internal/service/handler/spotify/import_artist_details.go:32]: yeah that makes sense. ill fix it. imma do the 2nd solution i think

Comment (aaronkim218) [reply backend/internal/service/handler/spotify/import_artist_details.go:55]: should be fixed now. i ran it once and it updated a bunch of ids and now when i run it, it returns an empty array. and there seems to only be 50 records missing photos now. i also checked the names of the artists and didn't recognize any of them so it checks out

**Comment by aaronkim218** [comment]
@ddusichka btw i changed http method to PATCH because i think it makes more sense and i have ocd

**Review by ddusichka — APPROVED** [review]
lgtm!

**Pull request merged**